sql/core/src/main/scala/org/apache/spark/sql/json/JSONRelation.scala

```diff
@@ -67,7 +67,6 @@ private[sql] class DefaultSource
         case SaveMode.Append =>
           sys.error(s"Append mode is not supported by ${this.getClass.getCanonicalName}")
         case SaveMode.Overwrite =>
-          fs.delete(filesystemPath, true)
```

Comment:
Why remove it? I think we can first get the file status (we also need to handle FileNotFoundException). Then, we check if data already exists. For overwrite, we delete it and if `delete` returns false and data does exist, we throw an exception.